Sources/SwiftProtobuf/Message+JSONAdditions.swift

```diff
@@ -93,3 +93,72 @@ public extension Message {
   }
 }
 
+
+public extension Array where Iterator.Element: Message {
```

Comment:
This is my biggest concern as well.  Other languages that support similar extension facilities (Javascript) have had real problems from folks extending common system classes this way.  And I see many, many people trying to add JSON extensions to core Swift classes, so this is a real possibility, especially for people transitioning from other systems to SwiftProtobuf.

Another option might be to add static methods to the Message type:
```
     static func arrayFrom(jsonUTF8Data: Data) throws -> [Self] { ... }

     static func jsonUTF8Data(array: [Self]) throws -> Data { ... }
```

Then folks could access this behavior as
```
    let decodedArray =  try MyMessage.arrayFrom(jsonUTF8Data: data);
    let jsonData = try MyMessage.jsonUTF8Data(array: someMessageArray);
```

This may be a little less obvious to some people, but would avoid the name-collision problem.

We could also provide generic functions in the `SwiftProtobuf` namespace rather than tying it into the Message type.

Thoughts?